metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -16,6 +16,7 @@ const uint64_t BLOCK_SIZE = 1 << 25;
 const uint64_t BUFFER_SIZE = 1024 * 1024; // 1 MiB
 const uint64_t ROW_REDUCTION_WIDTH = 32;
 const uint32_t MAX_NUM_FILES_OPEN = 2000;
+const uint64_t MAX_COLUMNS_IN_BATCH = 1'000'000;
```

Comment:
It would be nice to have this exposed in the CLI. That way, the users can have the option of keeping the thread count fixed and playing around with this accordingly.

Reply:
or ideally, having the batch size set automatically

Reply:
it's a different thing, not a batch size.

I just set this guy so that if there are too many tiny columns, we don't want to create millions of temp files, even though all of them would be small.

There are 4 different things you might be confusing:
1. trade-offs with threads when annotating (what you saw in the new trello card)
2. batch size, already set automatically -- we load as many annotation columns as we can fit in RAM.
3. number of open file descriptors (pre-merge chunks if there are too many of them, but throw error if we can't do the pre-merging in parallel, i.e. the number of threads is roughly 1000 or above.)
4. the number of temp files we'd be ok to create. For annotations with 50M columns, transforming them with disk-swap is hard because we can't keep 50M folders with chunks, that's inefficient, so I print warnings and errors if that happens.